coucharchive

```diff
@@ -74,7 +75,10 @@ def _couchdb_request(url):
     url = urlunsplit(parts)
     req = urllib.request.Request(url,
                                  headers={'Authorization': 'Basic %s' % auth})
-    response = urllib.request.urlopen(req)
+    if timeout is not None:
+        response = urllib.request.urlopen(req, timeout=timeout)
+    else:
+        response = urllib.request.urlopen(req)
```

Comment:
You can drop this change: in case `timeout` is `None`, these 2 lines are equivalent :wink:

Reply:
I wasn't sure xD.
Can you tell me how you find that the 2 lines are equivalent in the None case?
In the meantime here is the pushed fix!
```diff
--- a/coucharchive
+++ b/coucharchive
@@ -75,10 +75,7 @@ def _couchdb_request(url, timeout=None):
     url = urlunsplit(parts)
     req = urllib.request.Request(url,
                                  headers={'Authorization': 'Basic %s' % auth})
-    if timeout is not None:
-        response = urllib.request.urlopen(req, timeout=timeout)
-    else:
-        response = urllib.request.urlopen(req)
+    response = urllib.request.urlopen(req, timeout=timeout)
     return json.loads(response.read().decode('utf-8'))

Reply:
Actually I was wrong: it would be the case if the default value was `None` here (which is very usual, but not here): https://github.com/python/cpython/blob/9af0e47b1705457bb6b327c197f2ec5737a1d8f6/Lib/urllib/request.py#L139

I'm suspecting that using `None` is still correct, because of https://github.com/python/cpython/blob/9af0e47b1705457bb6b327c197f2ec5737a1d8f6/Lib/urllib/request.py#L507, but can't be sure. So, sorry... what I said was wrong!

Reply:
Ok, no issue, thanks for the links!
I restore the first version.